build_tools/update_environments_and_lock_files.py

```diff
@@ -90,6 +90,32 @@ def remove_from(alist, to_remove):
 
 
 build_metadata_list = [
+    {
+        "name": "pylatest_conda_forge_mkl_array-api_linux-64",
+        "type": "conda",
+        "tag": "gpu",
+        "folder": "build_tools/github",
+        "platform": "linux-64",
+        "channel": ["conda-forge", "pytorch", "nvidia"],
+        "conda_dependencies": common_dependencies
+        + [
+            "ccache",
+            # Make sure pytorch comes from the pytorch channel and not conda-forge
+            "pytorch::pytorch",
+            "pytorch-cuda",
+            "polars",
+            "pyarrow",
+            "cupy",
+            "array-api-compat",
+            "array-api-strict",
+        ],
+        "package_constraints": {
+            "blas": "[build=mkl]",
+            "pytorch::pytorch": "2.3.0",
+            "pytorch-cuda": "12.1",
+            "cupy": "13.1.0",
```

Comment:
I pinned them because I wanted version changes to be an explicit action. It was before I realised that we have a cron job that updates the lock file. I think for now this is good enough for me. Maybe one day in the future we should have a job with a minimum version number of these libraries that we want to support and a job that uses the latest version. But leaving that for the future, for now I unpinned them.